python/tests/operations/abs.py

```diff
@@ -0,0 +1,218 @@
+            "parameter",
+            data=np.array([[[-6, 1], [7, 2]], [[8, -3], [9, 4]]]),
+            samples=Labels(
+                ["sample", "positions"], np.array([[0, 1], [1, 1]], dtype=np.int32)
+            ),
+            components=[
+                Labels(["components"], np.array([[0], [1]], dtype=np.int32)),
+            ],
+        )
+        block_res2 = TensorBlock(
+            values=np.array([[1, 2], [3, 4], [5, 6]]),
+            samples=Labels(["samples"], np.array([[0], [2], [7]], dtype=np.int32)),
+            components=[],
+            properties=Labels(["properties"], np.array([[0], [1]], dtype=np.int32)),
+        )
+        block_res2.add_gradient(
+            "parameter",
+            data=np.array(
+                [
+                    [[10, -11], [12, -13]],
+                    [[14, -15], [-10, -11]],
+                    [[-12, 13], [14, 15]],
+                ]
+            ),
+            samples=Labels(
+                ["sample", "positions"],
+                np.array([[0, 1], [1, 1], [2, 1]], dtype=np.int32),
+            ),
+            components=[
+                Labels(["components"], np.array([[0], [1]], dtype=np.int32)),
+            ],
+        )
+        keys = Labels(
+            names=["key_1", "key_2"], values=np.array([[0, 0], [1, 0]], dtype=np.int32)
+        )
+        A = TensorMap(keys, [block_1, block_2])
+        tensor_abs = equistore.abs(A)
+        tensor_result = TensorMap(keys, [block_res1, block_res2])
+
+        assert equistore.equal(tensor_result, tensor_abs)
+
+    def test_self_abs_tensor_components(self):
+        A = equistore.load(
+            os.path.join(DATA_ROOT, "qm7-spherical-expansion.npz"), use_numpy=True
+        )
+
+        tensor_abs = equistore.abs(A)
+
+        abs_blocks = []
+        for _, block in A:
+            new_block = TensorBlock(
+                samples=block.samples,
+                components=block.components,
+                properties=block.properties,
+                values=np.abs(block.values),
+            )
+            sign_values = np.sign(block.values)
+            shape = ()
+            for c in block.components:
+                shape += (len(c),)
+            shape += (len(block.properties),)
+            for parameter, gradient in block.gradients():
+                diff_components = len(gradient.components) - len(block.components)
+                new_grad = gradient.data[:] * sign_values[
+                    gradient.samples["sample"]
+                ].reshape((-1,) + (1,) * diff_components + shape)
+                new_block.add_gradient(
+                    parameter,
+                    new_grad,
+                    gradient.samples,
+                    gradient.components,
+                )
+
+            abs_blocks.append(new_block)
+
+        tensor_result = TensorMap(A.keys, abs_blocks)
+        assert equistore.equal(tensor_result, tensor_abs)
+
+    def test_self_abs_tensor_complex(self):
```

Comment:
very nice, I didn't think about that, this is the first test with complex number i think, well done